File: client/src/app/dtos/post-todo-dto.ts
import { Priority } from "../enums/priority";

export class PostTodoDto {

    title: string;
    description: string;
    deadLineDate : Date;
    priority : Priority;

    constructor(_title: string,_description: string,_deadLineDate : Date,_priority : Priority){
    this.title = _title ;
    this.description = _description ;
    this.deadLineDate = _deadLineDate ;
    this.priority = _priority ;
    }
}

File: server/src/main/java/com/productivityapp/demo/domain/dto/PostTodoDto.java
package com.productivityapp.demo.domain.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.productivityapp.demo.domain.enumeration.Priority;

import java.time.LocalDate;

public record PostTodoDto(

        String title,
        
        String description,

        @JsonFormat(pattern = "dd/MM/yyyy")
        LocalDate deadLineDate,

        Priority priority


) {
}
